### PR TITLE
Add docs

### LINKED ISSUE #1: Add Docs
I haven't got around to setting up docs with Documenter.jl
It should be pretty standard setup

### PR DESCRIPTION
**Pull request by oxinabox**
Closes #1 

this is just a shell of docs to be filled in as features are added

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using ChainRulesDeclarationHelpers
+using Documenter
+using DocThemeIndigo
+using Markdown
```
Comment (nickrobinson251) [inline docs/make.jl]: why is this needed?

File: docs/Project.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[deps]
+ChainRulesDeclarationHelpers = "eeaa3d8b-e16a-5a7e-b044-1a854a548346"
+Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
```
Comment (nickrobinson251) [inline docs/Project.toml]: is this used?

**Review by nickrobinson251 — APPROVED** [review]

File: docs/Project.toml (hunk 2 above)
Comment (oxinabox) [reply docs/Project.toml]: It is not
```suggestion
```

File: docs/make.jl (hunk 1 above)
Comment (oxinabox) [reply docs/make.jl]: No idea.
I copied it from the ChainRulesOverloadGeneration docs.

Comment (oxinabox) [reply docs/make.jl]: Answer: it isn't
